### PR TITLE
Improve FreeBSD's Vagrant box

### PR DESCRIPTION
**Pull request by jlduran**
- x86_64-freebsd10 -> x86_64-freebsd-10
- Increase memory to 2048
- Upgrade installed packages
- Use the `git-lite` package

### CODE REVIEW

File: release/x86_64-freebsd10/Vagrantfile [hunk 1]
```diff
@@ -81,6 +81,9 @@ Vagrant.configure(2) do |config|
   #   sudo apt-get install -y apache2
   # SHELL
   config.vm.provision "shell", inline: <<-SHELL
-    pkg install -y bash gmake ruby rubygem-bundler git-subversion python2
+    freebsd-update --not-running-from-cron fetch install
```
Comment (ignisf) [inline release/x86_64-freebsd10/Vagrantfile]: Doesn't this get automatically run when you first start the box? I thought the box itself had an auto-upgrade script that was run on first boot?

File: release/x86_64-freebsd10/Vagrantfile [hunk 2]
```diff
@@ -81,6 +81,9 @@ Vagrant.configure(2) do |config|
...
+    pkg upgrade -y
```
Comment (ignisf) [inline release/x86_64-freebsd10/Vagrantfile]: Which packages get upgraded on first boot of the VM?

File: release/x86_64-freebsd10/Vagrantfile [hunk 3]
```diff
@@ -81,6 +81,9 @@ Vagrant.configure(2) do |config|
...
+    pkg install -y bash gmake ruby rubygem-bundler git-lite python2
+    gem update --system --no-document
```
Comment (ignisf) [inline release/x86_64-freebsd10/Vagrantfile]: Is there a specific issue which you are hitting with the RubyGems that come packaged from FreeBSD?

File: release/x86_64-freebsd10/Vagrantfile (hunk 1 above)
Comment (jlduran) [inline release/x86_64-freebsd10/Vagrantfile]: You are right! I'll remove this line.

File: release/x86_64-freebsd10/Vagrantfile (hunk 2 above)
Comment (jlduran) [inline release/x86_64-freebsd10/Vagrantfile]: None. The `firstboot-freebsd-update` updates the system, but the `firstboot-pkgs` installs a list of packages on the first boot, and for this box, the list is empty:

https://github.com/freebsd/freebsd/blob/release/10.3.0/release/tools/vagrant.conf#L8

At the time of this writing, this is the upgrade list:

    xkeyboard-config: 2.16 -> 2.17
    python27: 2.7.11_1 -> 2.7.12
    pixman: 0.32.8 -> 0.34.0
    perl5: 5.20.3_8 -> 5.20.3_13
    pciids: 20151224 -> 20160621
    libglesv2: 11.0.8 -> 11.2.2
    libglapi: 11.0.8 -> 11.2.2
    libedit: 3.1.20150325_1 -> 3.1.20150325_2,1
    libdrm: 2.4.65_1,1 -> 2.4.66,1
    libGL: 11.0.8 -> 11.2.2
    libEGL: 11.0.8 -> 11.2.2
    gettext-runtime: 0.19.6 -> 0.19.8.1
    gbm: 11.0.8 -> 11.2.2
    freetype2: 2.6.2 -> 2.6.3
    expat: 2.1.0_3 -> 2.1.1_2
    dri: 11.0.8,2 -> 11.2.2,2

File: release/x86_64-freebsd10/Vagrantfile (hunk 3 above)
Comment (jlduran) [inline release/x86_64-freebsd10/Vagrantfile]: Not really. I'll remove this one as well

**Comment by jlduran** [comment]
Will you be willing to accept a change to the directory name (`Gem::Platform.local.to_s`)?

    x86_64-freebsd10 -> x86_64-freebsd-10

**Comment by ignisf** [comment]
Hmmm, if we're to stick to the platform convention, shouldn't it be `amd64-freebsd-10`?

**Comment by jlduran** [comment]
I'm using `Gem::Platform.local.to_s` as the reference.

FreeBSD:

```ruby
irb(main):001:0> Gem::Platform.local.to_s
=> "x86_64-freebsd-10"
```

mac OS:

```ruby
irb(main):001:0> Gem::Platform.local.to_s
=> "x86_64-darwin-15"
```

Debian:

```ruby
irb(main):001:0> Gem::Platform.local.to_s
=> "x86_64-linux"
```

**Comment by ignisf** [comment]
Ok, I thought it was amd64 for some reason, then it's fine.

**Comment by jlduran** [comment]
👍 Thank you.

One last thing:

On 82492e5cf83f455d52c083b321c8cb2b5147486d, you destroy the box, any reason not to force it? (i.e. `vagrant box destroy -f`)

https://github.com/cowboyd/libv8/blob/82492e5cf83f455d52c083b321c8cb2b5147486d/Rakefile#L66

**Comment by ignisf** [comment]
Not that I can think of

**Pull request merged**